lib/spark.js

```diff
@@ -43,7 +43,7 @@ export default class Spark {
     const { indexerResult, provider } = await queryTheIndex(retrieval.cid)
     stats.indexerResult = indexerResult
 
-    if (indexerResult !== 'OK') return
+    if (indexerResult !== 'OK' && indexerResult !== 'HTTP_NOT_ADVERTISED') return
```

Comment:
```suggestion
    if (!(indexerResult === 'OK' || indexerResult === 'HTTP_NOT_ADVERTISED')) return
```

I find this more understandable: Skip if the indexer result isn't ok or http_not_advertised. Alternatively:

```suggestion
    if (!['OK', 'HTTP_NOT_ADVERTISED'].includes(indexerResult)) return
```

Reply:
Makes sense. I decided to extract a named variable to make this even easier to understand.

```js
    const providerFound = indexerResult === 'OK' || indexerResult === 'HTTP_NOT_ADVERTISED'
    if (!providerFound) return
```